ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/KnownPeers.hs

```diff
@@ -78,18 +78,14 @@ belowTarget actions
     -- We can only ask ones where we have not asked them within a certain time.
   , not (Set.null availableForPeerShare)
 
-    -- Are there any peers from the availableForPeerShare set we can really ask
-    -- (i.e. have correct PeerSharing permissions)?
-  , not (Set.null canAsk)
-
```

Comment:
Why was this removed? the `availableForPeerShare` set contains peers which might have `PeerSharingDisabled` so we can't issue requests to them.

Reply:
As I understand it it cannot, because we only add peers which enabled peer sharing to `availableForPeerShare`.

Reply:
I changed it so that `availableForPeerShare` only contains peers that actually support peersharing. This removes the need for `canAsk`.